### PR TITLE
Upgrade Spring Framework to 6.1.10

### PR DESCRIPTION
**Pull request by eshryane**
Revert downgrade to 6.1.4 in #1477

### CODE REVIEW

File: whois-commons/src/main/resources/applicationContext-commons.xml [hunk 1]
```diff
@@ -17,11 +17,11 @@
     <!-- @Transactional CAN be used on private methods with mode="aspectj". However, annotations on interfaces will NOT work! -->
     <tx:annotation-driven mode="aspectj"/>
 
-    <context:spring-configured/>
+    <context:spring-configured />
     <context:component-scan base-package="net.ripe.db.whois.common"/>
 
     <context:annotation-config/>
-    <cache:annotation-driven/>
+    <cache:annotation-driven mode="aspectj"/>
```
Comment (maggarwal13) [inline whois-commons/src/main/resources/applicationContext-commons.xml:24]: Can we use the annotation @EnableCaching instead?

Comment (eshryane) [reply whois-commons/src/main/resources/applicationContext-commons.xml:24]: Yes, according to http://www.springframework.org/schema/cache/spring-cache.xsd :
```
	Indicates that cache configuration is defined by annotations on bean classes,
	and that proxies are automatically to be created for the relevant annotated beans.

	The default annotations supported are Spring's @Cacheable, @CachePut and @CacheEvict.
	If spring-context-support and the JSR-107 API are on the classpath, additional proxies
	are automatically created for JSR-107 annotated beans, that is @CacheResult, @CachePut,
	@CacheRemove and @CacheRemoveAll.

	See org.springframework.cache.annotation.EnableCaching javadoc for information on
	code-based alternatives to this XML element.
```
But we should keep the "aspectj" mode declaration

File: whois-commons/src/main/resources/applicationContext-commons.xml [hunk 2]
```diff
@@ -17,11 +17,11 @@
     <!-- @Transactional CAN be used on private methods with mode="aspectj". However, annotations on interfaces will NOT work! -->
     <tx:annotation-driven mode="aspectj"/>
```
Comment (eshryane) [inline whois-commons/src/main/resources/applicationContext-commons.xml:18]: Also according to: http://www.springframework.org/schema/tx/spring-tx.xsd
We can replace the `tx:annotation-driven` element:
```
	See org.springframework.transaction.annotation.EnableTransactionManagement Javadoc
	for information on code-based alternatives to this XML element.
```

File: whois-commons/src/main/resources/applicationContext-commons.xml [hunk 3]
```diff
@@ -17,11 +17,11 @@
...
 
-    <context:spring-configured/>
+    <context:spring-configured />
```
Comment (eshryane) [inline whois-commons/src/main/resources/applicationContext-commons.xml:20]: How do we replace `context:spring-configured` ?

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 1 above)
Comment (maggarwal13) [reply whois-commons/src/main/resources/applicationContext-commons.xml:24]: Yes , @EnableCaching(mode = AdviceMode.ASPECTJ)?

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 3 above)
Comment (maggarwal13) [reply whois-commons/src/main/resources/applicationContext-commons.xml:20]: @EnableSpringConfigured

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 2 above)
Comment (maggarwal13) [reply whois-commons/src/main/resources/applicationContext-commons.xml:18]: We should be careful as there are multiple tranbsaction in whois, I remeber having seperate one for nrtmv4

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 1 above)
Comment (eshryane) [reply whois-commons/src/main/resources/applicationContext-commons.xml:24]: Yes sounds good

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 2 above)
Comment (eshryane) [reply whois-commons/src/main/resources/applicationContext-commons.xml:18]: I only see one `<tx:annotation-driven>` element in Whois now, maybe we fixed that?

Comment (maggarwal13) [reply whois-commons/src/main/resources/applicationContext-commons.xml:18]: We specify @Transactional(transactionManager = "nrtmTransactionManager") manually on reposiritry layer so all good

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 1 above)
Comment (eshryane) [reply whois-commons/src/main/resources/applicationContext-commons.xml]: Done

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 3 above)
Comment (eshryane) [reply whois-commons/src/main/resources/applicationContext-commons.xml]: Done

File: whois-commons/src/main/resources/applicationContext-commons.xml (hunk 2 above)
Comment (eshryane) [reply whois-commons/src/main/resources/applicationContext-commons.xml]: Done

**Review by maggarwal13 — APPROVED** [review]
Tested on prepdev : caching, transactional, shedlock and retry annotations

**Pull request merged**